crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -827,8 +854,13 @@ impl<'a, Provider: ResolverProvider> Resolver<'a, Provider> {
                 };
                 let package_id = dist.package_id();
 
-                // If the package does not exist in the registry, we cannot fetch its dependencies
-                if self.unavailable_packages.get(package_name).is_some() {
+                // If the package does not exist in the registry or locally, we cannot fetch its dependencies
+                if self.unavailable_packages.get(package_name).is_some()
+                    && self
+                        .installed_packages
+                        .get_packages(package_name)
+                        .is_empty()
+                {
                     debug_assert!(
                         false,
                         "Dependencies were requested for a package that is not available"
```

Comment:
How do we even get here btw?

Reply:
Maybe a dumb question since there's a `debug_assert!` here anyway.

Reply:
Ah if a version map cannot be retrieved for a package we track the package in `unavailable_packages` but that's no longer the sole way to have a candidate for a package so this is kind of a lazy way of allowing installed packages dependencies to be retrieved even if they were not available in the index.